Minesweeper.java

```diff
@@ -142,23 +200,23 @@ public Game(int gridSize, int bombAmount){
 
             // must be done after bomb population 
             // as bombs have a separate action listener
-            for (int y = 0; y < this.gridSize; y++) {
-                for (int x = 0; x < this.gridSize; x++) {
+            for (int y = 0; y < Game.gridSize; y++) {
+                for (int x = 0; x < Game.gridSize; x++) {
                     final Cell cell = cells[y][x];
-                    final Game self = this;
 
                     if (!cell.isBomb) {
+                        final int x2 = x;
+                        final int y2 = y;
                         cell.addActionListener(new ActionListener() {
                             @Override
                             public void actionPerformed(ActionEvent e) {
-                                self.computeNeighboringBombs(cell);
-                                cell.reveal();
+                                Game.computeNeighboringBombs(cell);
```

Comment:
note that this also means bringing back the `final Game self = this;` line
```suggestion
                                self.computeNeighboringBombs(cell);
```